contracts/protocol/facets/ExchangeHandlerFacet.sol

```diff
@@ -523,6 +561,61 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
         emit VoucherTransferred(exchange.offerId, _exchangeId, buyerId, msgSender());
     }
 
+    function onPremintedVoucherTransferred(
+        uint256 _tokenId,
+        address payable _to,
+        address _from,
+        address _rangeOwner,
+        address _sender
+    ) external override buyersNotPaused {
+        // Derive the offer id
+        uint256 offerId = _tokenId >> 128;
+
+        // Get the offer
+        (, Offer storage offer) = fetchOffer(offerId);
+
+        uint256 exchangeId = _tokenId & type(uint128).max;
+
+        // Cache protocol entities for reference
+        ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
+
+        if (offer.priceType == OfferPrice.Discovery) {
+            // Store the information about incoming voucher
+            ProtocolLib.ProtocolStatus storage ps = protocolStatus();
+
+            address priceDiscoveryContract = lookups.priceDiscoveryContractByVoucher[_tokenId];
+
+            if (ps.incomingVoucherCloneAddress != address(0)) {
+                ps.incomingVoucherId = _tokenId;
+
+                IBosonVoucher(ps.incomingVoucherCloneAddress).setCommitted(_tokenId, true);
+
+                commitToOfferInternal(_to, offer, exchangeId, true);
+            } else if (_from == priceDiscoveryContract && _to == _rangeOwner) {
+                delete lookups.priceDiscoveryContractByVoucher[_tokenId];
+            } else {
+                if (_from == _rangeOwner) {
+                    // @TODO: How to differente between when is seller depositing to AMM or a normal preminted token transfer (like opensea bid)
```

Comment:
I think I only now fully got what the problem is. What I wrote is true if users do, what we want them to do.

But yeah, what if somebody creates a priceDiscovery offer, lists it on openSea and just fulfils it directly via seaPort. Then we come to this place, and I believe that's what you had in mind when you wrote the comment.

Reply:
I couldn't think of any case.
I have implemented the check

Reply:
In the case of AMMs, sender !== to because AMM deposits are done through specific pool contracts for each pair. The sender in this case is the LSSVMPairFactory, while the to is a LSSVMPairERC20 contract.